app/views/admin/versions/_object_desc_and_link.html.haml

```diff
@@ -1,156 +1,162 @@
-  - last_deleted_version = PaperTrail::Version.where(item_type: version.item_type, item_id: version.item_id).last.reify
-  = last_deleted_version.try(:title) || last_deleted_version.try(:name)
+- when 'Role'
+  = 'role'
+  - role = current_or_last_object_state(version)
+  = link_if_alive version, role.name,
+          admin_conference_role_path(conference_id: Conference.find(version.conference_id).short_title, id: role.name)
+
+- when 'Venue'
+  = 'venue'
+  - venue = current_or_last_object_state(version)
+  = link_if_alive version, venue.name,
+          admin_conference_venue_path(conference_id: Conference.find(version.conference_id).short_title)
+
+- when 'Lodging'
+  = 'lodging'
+  - lodging = current_or_last_object_state(version)
+  = link_if_alive version, lodging.name,
+          admin_conference_lodgings_path(conference_id: Conference.find(version.conference_id).short_title)
+
+- when 'Room'
+  = 'room'
+  - room = current_or_last_object_state(version)
+  = link_if_alive version, room.name,
+          admin_conference_venue_rooms_path(conference_id: Conference.find(version.conference_id).short_title)
+
+- when 'Sponsor'
+  = 'sponsor'
+  - sponsor = current_or_last_object_state(version)
+  = link_if_alive version, sponsor.name,
+          admin_conference_sponsors_path(conference_id: Conference.find(version.conference_id).short_title)
+
+- when 'SponsorshipLevel'
+  = 'sponsorship level'
+  - sponsorship_level = current_or_last_object_state(version)
+  = link_if_alive version, sponsorship_level.title,
+          admin_conference_sponsorship_levels_path(conference_id: Conference.find(version.conference_id).short_title)
+
+- when 'Ticket'
+  = 'ticket'
+  - ticket = current_or_last_object_state(version)
+  = link_if_alive version, ticket.title,
+          admin_conference_ticket_path(conference_id: Conference.find(version.conference_id).short_title, id: version.item_id)
+
+- when 'Campaign'
+  = 'campaign'
+  - campaign = current_or_last_object_state(version)
+  = link_if_alive version, campaign.name,
+          admin_conference_campaigns_path(conference_id: Conference.find(version.conference_id).short_title)
+
+- when 'DifficultyLevel'
+  = 'difficulty level'
+  - difficulty_level = current_or_last_object_state(version)
+  = link_if_alive version, difficulty_level.title,
+          admin_conference_program_difficulty_level_path(conference_id: Conference.find(version.conference_id).short_title, id: version.item_id)
+
+- when 'Splashpage'
+  = link_if_alive version, 'splashpage',
+          admin_conference_splashpage_path(conference_id: Conference.find(version.conference_id).short_title)
+
+- when 'EmailSettings'
+  = link_if_alive version, 'email settings',
+          admin_conference_emails_path(conference_id: Conference.find(version.conference_id).short_title)
+
+- when 'User'
+  - if version.event == 'update'
+    = 'user'
+    = link_to User.find(version.item_id).name, admin_user_path(id: version.item_id)
```

Comment:
I've changed this, now it first check if the user exists, if not it will get the name from its last version.